addons/mail/models/mail_thread.py

```diff
@@ -267,7 +267,22 @@ def create(self, vals_list):
             for thread in threads:
                 subtype = thread._creation_subtype()
                 if subtype:  # if we have a subtype, post message to notify users from _message_auto_subscribe
-                    thread.sudo().message_post(subtype_id=subtype.id, author_id=self.env.user.partner_id.id)
+                    relations = subtype._get_auto_subscription_subtypes(self._name)[4]
+                    body = ''
+                    if relations:
+                        model_name = self.env['ir.model']._get(self._name).display_name
+                        parent_models = [
```

Comment:
I think this can be simplified a bit
```python
get_model = self.env["ir.model"]._get(p_model_name)
parent_models = [
    f'"{thread[fname].display_name}" ({get_model(p_model_name).display_name})'
    for p_model_name, f_names in relations.items()
    for f_name in f_names
    if thread[fname]
]
```

(maybe even more than this with functools :thinking: )

Reply:
Your proposition is great and I haven't found better with functools, thanks.